proto/src/main/java/com/google/gson/protobuf/LenientBase64Coder.java

```diff
@@ -0,0 +1,192 @@
+
+  /**
+   * Encodes a byte array into Base64 format. No blanks or line breaks are inserted in the output.
+   *
+   * @param in An array containing the data bytes to be encoded.
+   * @return A character array containing the Base64 encoded data.
+   */
+  public static char[] encode(byte[] in) {
+    return encode(in, 0, in.length);
+  }
+
+  /**
+   * Encodes a byte array into Base64 format. No blanks or line breaks are inserted in the output.
+   *
+   * @param in An array containing the data bytes to be encoded.
+   * @param iLen Number of bytes to process in <code>in</code>.
+   * @return A character array containing the Base64 encoded data.
+   */
+  public static char[] encode(byte[] in, int iLen) {
+    return encode(in, 0, iLen);
+  }
+
+  /**
+   * Encodes a byte array into Base64 format. No blanks or line breaks are inserted in the output.
+   *
+   * @param in An array containing the data bytes to be encoded.
+   * @param iOff Offset of the first byte in <code>in</code> to be processed.
+   * @param iLen Number of bytes to process in <code>in</code>, starting at <code>iOff</code>.
+   * @return A character array containing the Base64 encoded data.
+   */
+  public static char[] encode(byte[] in, int iOff, int iLen) {
+    int oDataLen = (iLen * 4 + 2) / 3; // output length without padding
+    int oLen = ((iLen + 2) / 3) * 4; // output length including padding
+    char[] out = new char[oLen];
+    int ip = iOff;
+    int iEnd = iOff + iLen;
+    int op = 0;
+    while (ip < iEnd) {
+      int i0 = in[ip++] & 0xff;
+      int i1 = ip < iEnd ? in[ip++] & 0xff : 0;
+      int i2 = ip < iEnd ? in[ip++] & 0xff : 0;
+      int o0 = i0 >>> 2;
+      int o1 = ((i0 & 3) << 4) | (i1 >>> 4);
+      int o2 = ((i1 & 0xf) << 2) | (i2 >>> 6);
+      int o3 = i2 & 0x3F;
+      out[op++] = map1[o0];
+      out[op++] = map1[o1];
+      out[op] = op < oDataLen ? map1[o2] : '=';
+      op++;
+      out[op] = op < oDataLen ? map1[o3] : '=';
+      op++;
+    }
+    return out;
+  }
+
+  /**
+   * Decodes a byte array from Base64 format. No blanks or line breaks are allowed within the Base64
+   * encoded input data.
+   *
+   * @param s A Base64 String to be decoded.
+   * @return An array containing the decoded data bytes.
+   * @throws IllegalArgumentException If the input is not valid Base64 encoded data.
+   */
+  public static byte[] decode(String s) {
+    return decode(s.toCharArray());
+  }
+
+  /**
+   * Decodes a byte array from Base64 format. No blanks or line breaks are allowed within the Base64
+   * encoded input data.
+   *
+   * @param in A character array containing the Base64 encoded data.
+   * @return An array containing the decoded data bytes.
+   * @throws IllegalArgumentException If the input is not valid Base64 encoded data.
+   */
+  public static byte[] decode(char[] in) {
+    return decode(in, 0, in.length);
+  }
+
+  /**
+   * Decodes a byte array from Base64 format. No blanks or line breaks are allowed within the Base64
```

Comment:
I think it's just that it accepts either standard or url encoded form, and trailing '=' chars are optional.